alby.go

```diff
@@ -52,12 +55,22 @@ func NewAlbyOauthService(svc *Service, e *echo.Echo) (result *AlbyOAuthService,
 }
 
 func (svc *AlbyOAuthService) FetchUserToken(ctx context.Context, app App) (token *oauth2.Token, err error) {
-	user := app.User
-	tok, err := svc.oauthConf.TokenSource(ctx, &oauth2.Token{
+	svc.tokenMutex.Lock()
+	defer svc.tokenMutex.Unlock()
+
+	user := User{}
+	err = svc.db.First(&user, &User{ID: app.UserId}).Error
+	currentToken := &oauth2.Token{
 		AccessToken:  user.AccessToken,
 		RefreshToken: user.RefreshToken,
 		Expiry:       user.Expiry,
-	}).Token()
+	}
+
+	if user.Expiry.After(time.Now()) {
```

Comment:
should we add 1 second to account for the network delay? that's how Alby JS SDK does it

Reply:
fixed